FAIL: TestCheckSuite/TestHealthChange

```#98  55.71 --- FAIL: TestCheckSuite/TestHealthChange (0.17s)
642	#98  55.71 suite.go:62: test panicked: runtime error: invalid memory address or nil pointer dereference
643	#98  55.71 goroutine 51 [running]:
644	#98  55.71 runtime/debug.Stack(0xc0000f58e0, 0x8c93a0, 0xd7b5a0)
645	#98  55.71 /toolchain/go/src/runtime/debug/stack.go:24 +0x9d
646	#98  55.71 github.com/stretchr/testify/suite.failOnPanic(0xc000242000)
647	#98  55.71 /go/pkg/mod/github.com/stretchr/testify@v1.3.1-0.20190311161405-34c6fa2dc709/suite/suite.go:62 +0x57
648	#98  55.71 panic(0x8c93a0, 0xd7b5a0)
649	#98  55.71 /toolchain/go/src/runtime/panic.go:522 +0x1b5
650	#98  55.71 github.com/talos-systems/talos/internal/app/machined/pkg/system/health_test.(*CheckSuite).TestHealthChange(0xc0001025e0)
651	#98  55.71 /src/internal/app/machined/pkg/system/health/health_test.go:98 +0x23f
652	#98  55.71 reflect.Value.call(0xc0001a35c0, 0xc0000f89b8, 0x13, 0x954509, 0x4, 0xc0000f5f30, 0x1, 0x1, 0xc0001d0668, 0x78, ...)
653	#98  55.71 /toolchain/go/src/reflect/value.go:447 +0x461
654	#98  55.71 reflect.Value.Call(0xc0001a35c0, 0xc0000f89b8, 0x13, 0xc0001d0730, 0x1, 0x1, 0x863119, 0x10, 0x0)
655	#98  55.71 /toolchain/go/src/reflect/value.go:308 +0xa4
656	#98  55.71 github.com/stretchr/testify/suite.Run.func2(0xc000242000)
657	#98  55.71 /go/pkg/mod/github.com/stretchr/testify@v1.3.1-0.20190311161405-34c6fa2dc709/suite/suite.go:133 +0x2fb
658	#98  55.71 testing.tRunner(0xc000242000, 0xc0001ac900)
659	#98  55.71 /toolchain/go/src/testing/testing.go:865 +0xc0
660	talos-systems/talos#98 55.71 created by testing.(*T).Run
```

https://ci.dev.talos-systems.io/talos-systems/talos/2580/1/16

--- a/internal/app/machined/pkg/system/health/health_test.go
+++ b/internal/app/machined/pkg/system/health/health_test.go
@@ -93,7 +93,13 @@ func (suite *CheckSuite) TestHealthChange() {
 		errCh <- health.Run(ctx, &settings, &state, check)
 	}()
 
-	time.Sleep(50 * time.Millisecond)
+	// wait for the first health change
+	for i := 0; i < 20; i++ {
+		if state.Get().Healthy != nil {
+			break
+		}
+		time.Sleep(50 * time.Millisecond)
+	}
 
 	suite.Require().False(*state.Get().Healthy)
 	suite.Require().Equal("health failed", state.Get().LastMessage)
@@ -138,7 +144,8 @@ func (suite *CheckSuite) TestCheckAbort() {
 		select {
 		case <-ctx.Done():
 			return ctx.Err()
-		case <-time.After(50 * time.Millisecond):
+		case <-time.After(50 * time.Second):
+			// should never be triggered, as Timeout is 1ms
 			return nil
 		}
 	}
@@ -152,7 +159,13 @@ func (suite *CheckSuite) TestCheckAbort() {
 		errCh <- health.Run(ctx, &settings, &state, check)
 	}()
 
-	time.Sleep(100 * time.Millisecond)
+	// wait for the first health change
+	for i := 0; i < 20; i++ {
+		if state.Get().Healthy != nil {
+			break
+		}
+		time.Sleep(50 * time.Millisecond)
+	}
 
 	suite.Require().False(*state.Get().Healthy)
 	suite.Require().Equal("context deadline exceeded", state.Get().LastMessage)
@@ -164,7 +177,7 @@ func (suite *CheckSuite) TestCheckAbort() {
 
 func (suite *CheckSuite) TestInitialState() {
 	var settings = health.Settings{
-		InitialDelay: time.Minute,
+		InitialDelay: 5 * time.Minute,
 	}
 
 	var state health.State
@@ -177,7 +190,7 @@ func (suite *CheckSuite) TestInitialState() {
 		errCh <- health.Run(ctx, &settings, &state, nil)
 	}()
 
-	time.Sleep(50 * time.Millisecond)
+	time.Sleep(100 * time.Millisecond)
 
 	suite.Require().Nil(state.Get().Healthy)
 	suite.Require().Equal("Unknown", state.Get().LastMessage)

